sources/packages/backend/workflow/test/2023-2024/parttime-assessment/costs/parttime-assessment-2023-2024-additional-transportation-costs.e2e-spec.ts

```diff
@@ -0,0 +1,339 @@
+      assessmentConsolidatedData.studentDataAdditionalTransportWeeks = 20;
+      assessmentConsolidatedData.studentDataAdditionalTransportCost = 80;
+      assessmentConsolidatedData.studentDataAdditionalTransportPlacement = true;
+      assessmentConsolidatedData.offeringDelivered =
+        OfferingDeliveryOptions.Online;
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      // calculatedDataAdditionalTransportationMax = min((94-25), 80) * min(30, 20) = 1380
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataAdditionalTransportationMax,
+      ).toBe(1380);
+      // calculatedDataTotalTransportationAllowance = min(1380, 80 * min(30, 20)) = 1380
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataTotalTransportationAllowance,
+      ).toBe(1380);
+    },
+  );
+
+  it(
+    "Should determine calculatedDataTotalTransportationAllowance when studentDataAdditionalTransportKm more than 280 " +
+      "and studentDataAdditionalTransportPlacement is false and offeringDelivered is onsite",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataAdditionalTransportKm = 300;
+      assessmentConsolidatedData.offeringWeeks = 30;
+      assessmentConsolidatedData.studentDataAdditionalTransportWeeks = 20;
+      assessmentConsolidatedData.studentDataAdditionalTransportCost = 80;
+      assessmentConsolidatedData.studentDataAdditionalTransportPlacement =
+        false;
+      assessmentConsolidatedData.offeringDelivered =
+        OfferingDeliveryOptions.Onsite;
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      // calculatedDataAdditionalTransportationMax = min(94, 80) * min(30, 20) = 1600
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataAdditionalTransportationMax,
+      ).toBe(1600);
+      // calculatedDataTotalTransportationAllowance = (30 - 20) * 13 + 1600 = 1730
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataTotalTransportationAllowance,
+      ).toBe(1730);
+    },
+  );
+
+  it(
+    "Should determine calculatedDataTotalTransportationAllowance when studentDataAdditionalTransportKm more than 280 " +
+      "and studentDataAdditionalTransportPlacement is false and offeringDelivered is online",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataAdditionalTransportKm = 300;
+      assessmentConsolidatedData.offeringWeeks = 30;
```

Comment:
I see that in all the test scenarios we are having `offering Weeks` >= `studentAdditionalTransportationWeeks`. Can we please have atleast one test checking `offering Weeks` < `studentAdditionalTransportationWeeks` scenario.

Reply:
I added a test scenario for `offering Weeks < studentAdditionalTransportationWeeks`. Please have a look.